internal/service/cloudbackupsnapshot/resource_cloud_backup_snapshot.go

```diff
@@ -123,20 +123,12 @@ func Resource() *schema.Resource {
 }
 
 func resourceCreate(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
-	conn := meta.(*config.MongoDBClient).Atlas
 	connV2 := meta.(*config.MongoDBClient).AtlasV2
 	groupID := d.Get("project_id").(string)
 	clusterName := d.Get("cluster_name").(string)
 
-	stateConf := &retry.StateChangeConf{
-		Pending:    []string{"CREATING", "UPDATING", "REPAIRING", "REPEATING"},
-		Target:     []string{"IDLE"},
-		Refresh:    cluster.ResourceClusterRefreshFunc(ctx, d.Get("cluster_name").(string), d.Get("project_id").(string), conn),
-		Timeout:    15 * time.Minute,
-		MinTimeout: 30 * time.Second,
-	}
-	_, err := stateConf.WaitForStateContext(ctx)
-	if err != nil {
+	stateConf := advancedcluster.CreateStateChangeConfig(ctx, connV2, groupID, clusterName, 15*time.Minute)
+	if _, err := stateConf.WaitForStateContext(ctx); err != nil {
```

Comment:
I know this is unrelated but very strange why we verify cluster state transition as part of creating a backup snapshot

Reply:
true, I can imagine it's to support when a user defines a cluster and a snapshot without depends_on between them